content/patterns/layered-zero-trust/lzt-secure-supply-chain.adoc

```diff
@@ -0,0 +1,557 @@
+
+[id="sign-container-image"]
+== Signing the image
+
+To sign the image, use RHTAS services again. For this step, you must also be logged in to the Quay service, because you write the digital signature to the repository.
+
+To reference the image you intend to sign, you must use its digest rather than its tag. You can obtain the digest of the image by using `skopeo` or by checking the Quay repository.
+
+.Procedure
+
+. Get the image digest:
++
+[source,terminal]
+----
+# Get the image digest
+# Add --tls-verify=false if needed
+$ export IMAGE_DIGEST="$(skopeo inspect docker://${IMAGE} --format '{{.Digest}}')"
+----
+
+. Set the image reference:
++
+[source,terminal]
+----
+# Set the IMAGE to be signed
+$ export IMAGE_REF="${QUAY_HOSTNAME}/ztvp/qtodo@${IMAGE_DIGEST}"
+----
+
+. Sign the image:
++
+[source,terminal]
+----
+# Sign the image
+$ cosign sign "${IMAGE_REF}" \
+  --fulcio-url="${FULCIO_URL}" \
+  --rekor-url="${REKOR_URL}" \
+  --identity-token="${OIDC_TOKEN}" \
+  --oidc-issuer "${OIDC_ISSUER_URL}" \
+  --yes
+----
+
+.Verification
+
+To verify the image, you can use `cosign` and check the output to see if it matches your identity and OIDC.
+
+* Verify the image by using the following command:
+
+[source,terminal]
+----
+# Verify the image signature
+$ cosign verify "${IMAGE_REF}" \
+  --rekor-url="${REKOR_URL}" \
+  --certificate-identity-regexp ".*${RHTAS_USER}.*" \
+  --certificate-oidc-issuer-regexp "${OIDC_ISSUER_URL}"
+----
+
+[id="generate-sbom"]
+== Generating the SBOM file
+
+The next step involves generating a Software Bill Of Materials (SBOM) for the image. This SBOM provides an inventory of the components that constitute the image.
+
+To generate this SBOM document, use an external tool called `syft`.
+
+.Procedure
+
+. Install and verify the `syft` tool:
++
+[source,terminal]
+----
+# Install syft
+$ curl -sSfL "https://get.anchore.io/syft" | sh -s -- -b /usr/local/bin
+
+# Verify installation
+$ syft version
+----
+
+. Use `syft` to generate the SBOM file for the image:
++
+[source,terminal]
+----
+# Generate the SBOM file
+$ syft scan registry:"${IMAGE}" -o spdx-json=qtodo-sbom.json
+----
+
+[id="attach-sbom"]
+== Attaching the SBOM to the image
+
+You must attach the generated SBOM file to the image, which enables verification by third parties if required. For this task, use the `cosign` tool again.
+
+Although direct attachment of the SBOM to the image is feasible, the recommended approach involves encapsulating the SBOM within a signed envelope, thereby generating an attestation. This method ensures the integrity of the SBOM, guaranteeing its provenance and confirming that it has not been subjected to unauthorized modifications.
+
+The process of signing and attaching the SBOM runs directly by using `cosign`, which operates on the image.
+
+.Procedure
+
+* Attach the SBOM to the image:
++
+[source,terminal]
+----
+# Attach the SBOM
+$ cosign attest \
```

Comment:
Is there a reason we use the OIDC token obtained previously in prior cosign commands but now we use the authorization code flow with an actual user?